### PR TITLE
Feature/external booking time #75

### PR DESCRIPTION
**Pull request by va-dym-ko-val**
image
image

### CODE REVIEW

File: frontend/src/app/modules/external-booking/external-booking-choose-meeting-page/external-booking-choose-meeting-page.component.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<h1>Please, choose meeting</h1>
+<div class="meeting-cards">
+    <ng-template
+        *ngIf="selectedUserAvailabilitySlots.userSlots.length !== 0; else noUserSlots"
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-choose-meeting-page/external-booking-choose-meeting-page.component.html]:
```suggestion
        *ngIf="selectedUserAvailabilitySlots.userSlots.length; else noUserSlots"
```

File: frontend/src/app/modules/external-booking/external-booking-choose-meeting-page/external-booking-choose-meeting-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    public selectedUserAvailabilitySlots: IUserPersonalAndTeamSlots;
+
+    @Output() selectedDurationAndLocationEvent = new EventEmitter<{ duration: number; location: LocationType }>();
+
+    public LocationType = LocationType;
+
+    public LocationTypeToLabelMapping = LocationTypeToLabelMapping;
+
+    // eslint-disable-next-line no-empty-function
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-choose-meeting-page/external-booking-choose-meeting-page.component.ts]: Please delete eslint-disable-next-line
And use BaseComponent

File: frontend/src/app/modules/external-booking/external-booking-choose-meeting-page/external-booking-choose-meeting-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    constructor(public spinnerService: SpinnerService, private availabilitySlotService: AvailabilitySlotService) {}
+
+    ngOnInit(): void {
+        this.availabilitySlotService.getUserPersonalAndTeamSlots(this.selectedUserId).subscribe((slots) => {
+            this.selectedUserAvailabilitySlots = slots;
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-choose-meeting-page/external-booking-choose-meeting-page.component.ts]: You need to unsubscribe

File: frontend/src/app/modules/external-booking/external-booking-choose-time-page/external-booking-choose-time-page.component.html [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                <h3>{{ this.calendarWeek.firstDay.getTime() + i * 24 * 60 * 60 * 1000 | date: 'dd' }}</h3>
+            </div>
+        </div>
+        <ng-container *ngFor="let in of slotsCount; let i = index">
+            <ng-container *ngFor="let day of daysWithTimeRange; let j = index">
+                <ng-container
+                    *ngIf="
+                        isDateInRange(theEarliestStartOfTimeRanges, day.startTime, day.finishTime, i);
+                        else defaultTimeSlot
+                    ">
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-choose-time-page/external-booking-choose-time-page.component.html]:
```suggestion
                        else defaultTimeSlot">
```

File: frontend/src/app/modules/external-booking/external-booking-choose-time-page/external-booking-choose-time-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+    public nowDate: Date = new Date(Date.now());
+
+    constructor(
+        public spinnerService: SpinnerService,
+        private availabilitySlotService: AvailabilitySlotService,
+        private route: ActivatedRoute,
+    ) {
+        this.availabilitySlotService.getUserPersonalAndTeamSlots(this.selectedUserId).subscribe((slots) => {
+            this.selectedUserAvailabilitySlots = slots;
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-choose-time-page/external-booking-choose-time-page.component.ts]: See the previous comment

File: frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts [hunk 6]
```diff
@@ -8,13 +11,28 @@ import { UserService } from '@core/services/user.service';
...
+export class ExternalBookingPageComponent implements OnInit {
+    public menu: IExternalBookingSideMenu = {} as IExternalBookingSideMenu;
+
+    public LocationTypeToLabelMapping = LocationTypeToLabelMapping;
+
+    public LocationType = LocationType;
 
     // eslint-disable-next-line no-empty-function
-    constructor(public spinnerService: SpinnerService, private userService: UserService) {
+    constructor(public spinnerService: SpinnerService, private userService: UserService, public router: Router) {}
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts]: See previous comments

File: frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts [hunk 7]
```diff
@@ -8,13 +11,28 @@ import { UserService } from '@core/services/user.service';
...
+
+    ngOnInit(): void {
         this.userService.getCurrentUser().subscribe((user) => {
-            this.selectedUser = user;
+            this.menu.user = user;
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts]:
```suggestion
            this.menu = {
                ...this.menu,
                user: user
            };
```

File: frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts [hunk 8]
```diff
@@ -8,13 +11,28 @@ import { UserService } from '@core/services/user.service';
...
         });
     }
+
+    public addDurationAndLocationInMenu(data: any): void {
+        this.menu.duration = data.duration;
+        this.menu.location = data.location;
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts]: see the previous comment

File: frontend/src/app/modules/external-booking/external-booking-routing.module.ts [hunk 9]
```diff
@@ -2,12 +2,29 @@ import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 import { AuthGuard } from '@core/guards/auth.guard';
 
+// eslint-disable-next-line max-len
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-routing.module.ts]: Do not use eslint-disable-next-line

File: frontend/src/app/modules/external-booking/external-booking.module.ts [hunk 10]
```diff
@@ -3,11 +3,18 @@ import { NgModule } from '@angular/core';
 import { MaterialModule } from '@shared/material/material.module';
 import { SharedModule } from '@shared/shared.module';
 
+// eslint-disable-next-line max-len
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking.module.ts]: do not use eslint-disable-next-line

File: frontend/src/styles/material.sass [hunk 11]
```diff
@@ -117,3 +117,7 @@ snack-bar-container
 .mat-form-field-infix
     padding: 0
     border-top: 0
+
+.mat-form-field
+    border: 2px solid #BEC5D0
```
Comment (tatianahutii) [inline frontend/src/styles/material.sass]: Move color to shared file

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by tatianahutii** [comment]
Please align items
image

**Comment by va-dym-ko-val** [comment]
image

File: frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts [hunk 12]
```diff
@@ -8,10 +12,35 @@ import { UserService } from '@core/services/user.service';
...
+    ngOnInit(): void {
+        this.userService.getCurrentUser().subscribe((user) => {
+            this.menu = {
+                ...this.menu,
+                user,
+            };
+        });
+    }
+
+    public addDurationAndLocationInMenu(data: any): void {
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts]: Please do not use any

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]
LGTM

**Pull request merged**